Fixed bug where key was never being sent to Google

aj-mcgowan-dufl opened the pull request:
Keys were not being sent to Google unless it was a signed request...this isn't a very elegant solution but it works and it's non invasive...

vivet commented:
If I remember correctly, its not in all cases that the key should be send. I will check the PR soon, just need to make sure if i am right or not

aj-mcgowan-dufl commented:
Cool thanks...the way I read it is that the key is always optional...the issue we were running into by the way was over-running the free tier of requests...everything looked like it was working fine before, but when we over-ran the free tier on timezone requests we were getting over query limit errors since even though we were setting the key, it wasn't being passed through to the request.

vivet commented:
And you are sure that the query over limit errors wasn't because of too frequent requests. I have experienced that.

aj-mcgowan-dufl commented:
Yeah, we ran into that as well but we have rate limiting/retry logic built in (something that I'd love to add to your library if you are open to it). But having a paid account (and thus needing the key) increases the rate limit as well as the total requests per 24 hour period (and starts billing for overage rather than rejecting requests). I implemented this change by the way and then swapped the binaries out vs the nuget package version and it resolved the issue 100% and we were then also able to see the number of requests on the Google API console.

vivet commented:
The public key should mean anything, except statistics about quota. I will look into it anyway and let you know

vivet commented:
You should instead make this method invoke its base. 
https://github.com/vivet/GoogleApi/blob/master/GoogleApi/Entities/Common/SignableRequest.cs#L59

That if the whole maps api supports the same as time zone, otherwise add the key in the time zone requst method of same name.

aj-mcgowan-dufl commented:
When we just added the key, it was being ignored because we weren't doing a fully signed request with a client id as well...that's why I added this change...

Foxxloic commented:
I agree with @aj-mcgowan-dufl, the key need to be sent when filled.

vivet commented:
I have added the Key in the SignedRequest instead. But if filled it should now be added to the url.
I haven't made nuget package yet. but will soon. 
Let me know if you still have issues.
